cmd/publicize/server.go

```diff
@@ -0,0 +1,237 @@
+			if err := s.handleIssueComment(l, ic); err != nil {
+				s.log.WithError(err).WithFields(l.Data).Info("Failed to publicize pull request.")
+			}
+		}()
+	default:
+		logrus.Debugf("skipping event of type %q", eventType)
+	}
+	return nil
+}
+
+func (s *server) handleIssueComment(l *logrus.Entry, ic github.IssueCommentEvent) error {
+	if !ic.Issue.IsPullRequest() || ic.Action != github.IssueCommentActionCreated {
+		return nil
+	}
+
+	if !publicizeRe.MatchString(ic.Comment.Body) {
+		return nil
+	}
+
+	org := ic.Repo.Owner.Login
+	repo := ic.Repo.Name
+	num := ic.Issue.Number
+	commentAuthor := ic.Comment.User.Login
+
+	l = l.WithFields(logrus.Fields{
+		github.OrgLogField:  org,
+		github.RepoLogField: repo,
+		github.PrLogField:   num,
+	})
+
+	logger := s.log.WithFields(l.Data)
+
+	// Only org members should be able to publicize a pull request.
+	ok, err := s.ghc.IsMember(org, commentAuthor)
+	if err != nil {
+		return err
+	}
+	if !ok {
+		resp := fmt.Sprintf("only [%s](https://github.com/orgs/%s/people) org members may request a publicize", org, org)
+		logger.Info(resp)
+		return s.ghc.CreateComment(org, repo, num, plugins.FormatICResponse(ic.Comment, resp))
+	}
+
+	logger.Info("Publicize of PR has been requested.")
+
+	pr, err := s.ghc.GetPullRequest(org, repo, num)
+	if err != nil {
+		return err
+	}
+	baseBranch := pr.Base.Ref
+
+	var prCommits []string
+	commits, err := s.ghc.ListPRCommits(org, repo, num)
+	if err != nil {
+		return err
+	}
+
+	for _, commit := range commits {
+		prCommits = append(prCommits, commit.SHA)
+	}
+
+	// publicize only merged PRs.
+	if !pr.Merged {
+		resp := "cannot publicize an unmerged PR"
+		logger.Info(resp)
+		return s.ghc.CreateComment(org, repo, num, plugins.FormatICResponse(ic.Comment, resp))
+	}
+
+	logger.Infof("searching bond for %s/%s", org, repo)
+	bondedOrgRepo, ok := s.bonds[fmt.Sprintf("%s/%s", org, repo)]
+	if !ok {
+		logger.Warn("There is no bond for the current repository.")
+		resp := "cannot publicize because there is no bond for the current pull request"
+		return s.ghc.CreateComment(org, repo, num, plugins.FormatICResponse(ic.Comment, resp))
+	}
+
+	logger.Infof("Trying to merge the PR to destination: %s/%s@%s", org, repo, baseBranch)
+	if err := s.mergeAndPushToRemote(logger, bondedOrgRepo, num, org, repo, baseBranch, s.confirm, prCommits); err != nil {
+		return fmt.Errorf("couldn't merge the pull request and push to the destination: %v", err)
+	}
+
+	return s.ghc.CreateComment(org, repo, num, plugins.FormatICResponse(ic.Comment, "Publicize suceeded."))
+}
+
+func (s *server) mergeAndPushToRemote(logger *logrus.Entry, bondedOrgRepo string, prNumber int, org, repo, branch string, confirm bool, commits []string) error {
+	// this is already validated
+	bondedOrg := strings.Split(bondedOrgRepo, "/")[0]
+	bondedRepo := strings.Split(bondedOrgRepo, "/")[1]
+
+	bondURIRaw := fmt.Sprintf("https://github.com/%s/%s", bondedOrg, bondedRepo)
+	bondURI, err := url.Parse(bondURIRaw)
```

Comment:
This belongs in a GitHub client library

Reply:
I just want to construct the uri with the user in order to use it in the git commands.